h/api/search/query.py

```diff
@@ -97,29 +96,24 @@ def build(request_params, userid=None, search_normalized_uris=False):
     }
 
 
-def _match_clause_for_uri(uristr):
+def _filter_for_uri_match(uristr):
     """Return an Elasticsearch match clause dict for the given URI."""
     uristrs = uri.expand(uristr)
-    matchers = [{"match": {"uri": u}} for u in uristrs]
+    clauses = [{"match": {"uri": u}} for u in uristrs]
 
-    if len(matchers) == 1:
-        return matchers[0]
-    return {
-        "bool": {
-            "minimum_should_match": 1,
-            "should": matchers
-        }
-    }
+    if len(clauses) == 1:
+        return {"query": clauses[0]}
+    return {"query": {"bool": {"should": clauses}}}
 
 
-def _term_clause_for_uri(uristr):
+def _filter_for_uri_term(uristr):
     """Return an Elasticsearch term clause for the given URI."""
     uristrs = uri.expand(uristr)
-    filters = [{"term": {"target.source_normalized": uri.normalize(u)}}
+    clauses = [{"term": {"target.source_normalized": uri.normalize(u)}}
                for u in uristrs]
 
-    if len(filters) == 1:
-        return filters[0]
+    if len(clauses) == 1:
+        return clauses[0]
     return {
-        "or": filters
+        "or": clauses
```

Comment:
I know it's from the original URI normalization pr, but use of or here seems off to me. Why not use terms? Rreturn {"terms": {"target.source_normalized": [<uri_strings>]}}. This does or logic. And when there's only one URI string then just {"term": {"target.source_normalized": <uri_string>}}